python/src/exactextract/writer.py

```diff
@@ -129,6 +129,169 @@ def features(self):
             return pd.DataFrame(self.fields, copy=False)
 
 
+class QGISWriter(Writer):
+    """Creates QGSVectorLayer"""
+
+    def __init__(self, *, srs_wkt=None):
+        super().__init__()
+        from qgis.core import QgsCoordinateReferenceSystem
+
+        self.fields = []
+        self.feats = []
+        self.geom_type = "unknown"
+        self.fields_set = False
+
+        self.crs = QgsCoordinateReferenceSystem()
+        self.crs.createFromWkt(srs_wkt)
+
+    def add_operation(self, op):
+        self.fields.append(op.name)
+
+    def add_column(self, col_name):
+        self.fields.append(col_name)
+
+    def write(self, feature):
+        import json
+
+        from qgis.core import QgsFeature, QgsJsonUtils
+
+        f = JSONFeature()
```

Comment:
I would suggest writing a class to wrap `QgsFeature` (see `GDALFeature` for an example). Then you can avoid the intermediate step of creating a `JSONFeature`.